### PR TITLE
update recipe files for the IBMc test runs.

### PR DESCRIPTION
**Pull request by sunilkumarn417**
Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

- update test run results in RH recipe files from post-results jobs.

### CODE REVIEW

File: pipeline/post-results.groovy [hunk 1]
```diff
@@ -66,6 +66,30 @@ node(nodeName) {
...
+            def rhcsVersion = sharedLib.getRHCSVersionFromArtifactsNvr()
+            majorVersion = rhcsVersion["major_version"]
+            minorVersion = rhcsVersion["minor_version"]
+
+            def latestContent = sharedLib.readFromReleaseFile(
+                    majorVersion, minorVersion
+                )
+
+            if ( latestContent.containsKey(tierLevel) ) {
+                latestContent[tierLevel] = composeInfo
```
Comment (psathyan) [inline pipeline/post-results.groovy:85]: Shouldn't the `composeInfo` hold both RH build location information along with IBM-C location

File: pipeline/unsigned-container-image-listener.groovy [hunk 2]
```diff
@@ -90,7 +90,9 @@ node(nodeName) {
                 "email": "ceph-qe@redhat.com"
             ],
             "build": [
-                "repository": compose.repository
+                "repository": compose.repository,
+                "composes": releaseDetails.latest.composes,
+                "version": cephVersion
```
Comment (psathyan) [inline pipeline/unsigned-container-image-listener.groovy:95]: Would it make sense to put this information under `redhat`? that is
```
"build": [
  "redhat": [
    "repository": compose.repository,
    "composes": releaseDetails.latest.composes,
    "version": cephVersion
  ]
]
```

File: pipeline/ibm-tier-0.groovy [hunk 3]
```diff
@@ -164,7 +167,8 @@ node(nodeName) {
                 wait: false,
                 job: "tier-x",
                 parameters: [string(name: 'rhcephVersion', value: rhcephVersion),
-                            string(name: 'buildType', value: buildPhase)]
+                            string(name: 'buildType', value: buildPhase),
+                            string(name: 'CiMessageMap' value: buildArtifact)]
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply pipeline/ibm-tier-0.groovy:171]: Done

File: pipeline/ibm-tier-0.groovy [hunk 4]
```diff
@@ -44,6 +45,7 @@ node(nodeName) {
         /* Prepare pipeline stages using RHCEPH version */
         rhcephVersion = "${params.rhcephVersion}" ?: ""
         buildType = "${params.buildType}" ?: ""
+        buildArtifact = "${params.CiMessageMap}" ?: ""
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply pipeline/ibm-tier-0.groovy:48]: Done

File: pipeline/post-results.groovy [hunk 5]
```diff
@@ -66,6 +66,30 @@ node(nodeName) {
 
         //Remove the sync results folder
         sh script: "rclone purge ${remoteName}:${reportBucket}/${resultDir}"
+
+        // Update RH recipe file
+        def testStatus = msgMap["test"]["result"]
+        def composeInfo = readJSON text: msgMap["build"]
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply pipeline/post-results.groovy]: Yes, we don't need to read here. Addressed map issue at ibm-jobs itself.

File: pipeline/post-results.groovy (hunk 1 above)
Comment (sunilkumarn417) [reply pipeline/post-results.groovy:85]: As discussed, not needed.

File: pipeline/unsigned-container-image-listener.groovy (hunk 2 above)
Comment (sunilkumarn417) [reply pipeline/unsigned-container-image-listener.groovy:95]: As discussed, not needed.

**Review by psathyan — COMMENTED** [review]
Looks good... we would need the change in sendEmail to get the container image information. Would you be able include that change also?

**Review by psathyan — APPROVED** [review]

**Review by anrao19 — APPROVED** [review]

**Pull request merged**